Add a documentation note about AWS_PROFILE (#5931)
nextflow-io/nextflow: docs/amazons3.md

1. The `nextflow.config` file in the pipeline execution directory
2. The environment variables `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`
3. The environment variables `AWS_ACCESS_KEY` and `AWS_SECRET_KEY`

> 4. A profile in the AWS credentials file located at `~/.aws/credentials` (uses the environment variable `AWS_PROFILE` if set, otherwise the `default` profile)
> 5. A profile in the AWS client configuration file located at `~/.aws/config` (uses the environment variable `AWS_PROFILE` if set, otherwise the `default` profile)

Reviewer: ```suggestion
4. The profile in the AWS credentials file located at `~/.aws/credentials`
    - Uses the `default `profile or the environment variable `AWS_PROFILE` if set
5. The profile in the AWS client configuration file located at `~/.aws/config`
    - Uses the `default `profile or the environment variable `AWS_PROFILE` if set
```
